### ISSUE TITLE
Integration with Connectwise Agreement Billing

### ISSUE THREAD

**Issue by dsibiski** [AutoElevate/roadmap#36]

**Comment by Lowell-Syncrety** [comment]
Yes! This would be very useful for us as well. The current invoice has no customer information, so we have to provide manual access for the person reconciling products each month so they can look at, and expand each customer to see how many they have. They're not technical staff so they don't normally have (nor need) access.

**Comment by mendykrauss** [comment]
Any update on this?

**Comment by JImHaywood90** [comment]
This would be very useful. Currently, we use the Connectwise Automate API to compile a device count for each client by looking for windows workstations with AutoElevate installed, which we then use to update the clients agreement in Manage using the CWM Rest API. All working fine, but the script takes a very long time to run!

**Comment by qcomer** [comment]
@JImHaywood90 Why not use the Managed Services Agreement in Automate to sync? Built in and doesn’t require a script.

**Comment by JImHaywood90** [comment]
@qcomer  I'm not entirely sure what you are referring to here. Can you provide a link to the documentation for this?

**Comment by qcomer** [comment]
And, by all means...I am not saying this should be the end solution....but, it is a workaround much simpler than what some others mentioned they are currently doing.

@dsibiski  This should be high up on the list.

**Comment by dsibiski** [comment]
> And, by all means...I am not saying this should be the end solution....but, it is a workaround much simpler than what some others mentioned they are currently doing.
> 
> @dsibiski  This should be high up on the list.

@qcomer Noted. 👍

**Comment by Lowell-Syncrety** [comment]
@qcomer Wow. Thanks for that! It looks interesting... as far as I can see, however, in Agreement Mapping it looks like you can only map one Managed Service Template per customer. Or is that a psuedo multi-select field? For example, if I wanted to also do this for another product not just AutoElevate.

**Comment by Virtual-Nomad** [comment]
Cleaver work around for those running automate.... we just run CW manage and the billing for AE is somewhat of an issue to track manually. We desperately need direct billing integration.
